### PR TITLE
[PCRE2_jll] Upgrade to 10.42

### LINKED ISSUE #47936: Regex fails expected match
I created a minimal working example for a problem in the regular expression engine. I was able to reproduce the issue on the latest stable release 1.8.3, on 1.9.0-alpha1, and on master built today. It seems to be a problem with the engine failing to backtrack.

```julia
julia> match(r"a+[bc]+c", "ababc")

julia> versioninfo()
Julia Version 1.10.0-DEV.156
Commit 9be3c85e49 (2022-12-19 21:17 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: 72 × Intel(R) Xeon(R) Gold 6150 CPU @ 2.70GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-14.0.6 (ORCJIT, skylake-avx512)
  Threads: 1 on 72 virtual cores
```

```julia
julia> match(r"a+[bc]+c", "ababc")

julia> versioninfo()
Julia Version 1.8.3
Commit 0434deb161 (2022-11-14 20:14 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: 72 × Intel(R) Xeon(R) Gold 6150 CPU @ 2.70GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-13.0.1 (ORCJIT, skylake-avx512)
  Threads: 1 on 72 virtual cores
```

### PR DESCRIPTION
**Pull request by Seelengrab**
Fixes #47936

I followed the usual memo by @giordano :

 * update the version number in `stdlib/PCRE2_jll/Project.toml`
 * update the version number in tests `stdlib/PCRE2_jll/test/runtests.jl`
 * update the version number in `deps/pcre.version`
 * refresh checksums with `make -f contrib/refresh_checksums.mk -j pcre`

Is the number of checksums there correct? Seems like there's only one now :thinking:

### CODE REVIEW

**Comment by giordano** [comment]
> Is the number of checksums there correct?

No, the name to pass to the Makefile is probably `pcre2`

**Comment by Seelengrab** [comment]
```
[sukera@tower julia]$ make -f contrib/refresh_checksums.mk -j pcre2
make: *** No rule to make target 'pcre2'.  Stop.
[sukera@tower julia]$ make -f contrib/refresh_checksums.mk -j PCRE2
make: *** No rule to make target 'PCRE2'.  Stop.
[sukera@tower julia]$ make -f contrib/refresh_checksums.mk -j PCRE
make: *** No rule to make target 'PCRE'.  Stop.
```

It does not like either of these, only `pcre` :/

**Comment by giordano** [comment]
Did it download any files at all? Try removing the `-j` if you have race conditions

**Comment by Seelengrab** [comment]
That did it - ought to work now.

**Comment by KristofferC** [comment]
Maybe add the test from https://github.com/JuliaLang/julia/issues/47936 as well?

File: stdlib/PCRE2_jll/test/runtests.jl [hunk 1]
```diff
@@ -6,5 +6,16 @@ using Test, Libdl, PCRE2_jll
...
+@testset "#47936" begin
+    tests = (r"a+[bc]+c",
+             r"a+[bc]{1,2}c",
+             r"(a)+[bc]+c",
+             r"a{1,2}[bc]+c",
+             r"(a+)[bc]+c")
+    for re in tests
+        @test !isnothing(match(re, "ababc"))
+    end
+end
```
Comment (giordano) [inline stdlib/PCRE2_jll/test/runtests.jl:21]: Missing newline
```suggestion
end

```

**Comment by martinholters** [comment]
Backport to 1.6 as well, given that the upgrade that caused #47936 was backported (between 1.6.6 and 1.6.7)?

**Pull request merged**